### PR TITLE
Improve route comparator interface

### PR DESCRIPTION
**Pull request by kmadsen**
### Description

Proposing an improvement to the internal RouteComparator interface as we consider making it public.

### CODE REVIEW

**Comment by kmadsen** [comment]
To make this `RouteComparator` public. Considering actual locations to put it.

Right now it is local to where it is used, near faster route, but it would make sense to put it in a higher package since it does not only apply to faster route.

- [ ] Keep in com.mapbox.navigation.core.fasterroute
- [ ] Create new package, but i'm not coming up with good names
- [ ] Keep thinking

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/RouteComparator.kt [hunk 1]
```diff
@@ -15,10 +15,10 @@ internal class RouteComparator {
      * @param routeProgress current route progress
      * @param alternativeRoute suggested new route
      *
-     * @return true when the alternative route has different
-     * geometry from the current route progress
+     * @return true when the alternative is the same route,
+     * false if it is a new route.
```
Comment (LukasPaczos) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/RouteComparator.kt:19]:
```suggestion
     * false if it is a different route.
```

**Review by LukasPaczos — APPROVED** [review]
Let's merge regardless of whether we want to make it public.

**Comment by Guardiola31337** [comment]
Why `skip changelog`? 🤔

**Comment by LukasPaczos** [comment]
The API is internal.

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteDetector.kt [hunk 2]
```diff
@@ -16,7 +16,7 @@ internal class FasterRouteDetector(
         val alternativeDuration = alternativeRoute.duration()
         val weightedDuration = routeProgress.durationRemaining * PERCENTAGE_THRESHOLD
         val isRouteFaster = alternativeDuration < weightedDuration
-        return@withContext isRouteFaster && routeComparator.isRouteDescriptionDifferent(
+        return@withContext isRouteFaster && routeComparator.isSameRoute(
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteDetector.kt:19]: It seems we inverted the logic but we didn't add `!routeComparator.isSameRoute` `!`. Did we miss it?

Comment (kmadsen) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteDetector.kt:19]: 😱  yeah

Comment (kmadsen) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteDetector.kt]: wow thanks and nice catch!

negating and renaming broke the tests inside `FasterRouteDetectorTest`. so it wasn't caught from the unit tests i was relying on 😅 . all updated

**Pull request merged**